cli/command/idresolver/idresolver.go

```diff
@@ -45,6 +46,22 @@ func (r *IDResolver) get(ctx context.Context, t interface{}, id string) (string,
 			return id, nil
 		}
 		return service.Spec.Annotations.Name, nil
+	case swarm.Task:
+		task := t.(swarm.Task)
+		// If the caller passes the full task there's no need to do a lookup.
+		if task.ID == "" {
+			var err error
+
+			task, _, err = r.client.TaskInspectWithRaw(ctx, id)
+			if err != nil {
+				return id, nil
+			}
+		}
+		service, err := r.Resolve(ctx, swarm.Service{}, task.ServiceID)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s.%d.%s", service, task.Slot, stringid.TruncateID(task.ID)), nil
```

Comment:
Seems like now that this is being generalized, it should handle tasks that are not associated with a service.